src/pudl/output/pudltabl.py

```diff
@@ -129,6 +130,79 @@ def __init__(
+            "demand_side_management_ee_dr_eia861": "demand_side_management_ee_dr_eia861",
+            "demand_side_management_misc_eia861": "demand_side_management_misc_eia861",
+            "distributed_generation_tech_eia861": "distributed_generation_tech_eia861",
+            "distributed_generation_fuel_eia861": "distributed_generation_fuel_eia861",
+            "distributed_generation_misc_eia861": "distributed_generation_misc_eia861",
+            "distribution_systems_eia861": "distribution_systems_eia861",
+            "dynamic_pricing_eia861": "dynamic_pricing_eia861",
+            "energy_efficiency_eia861": "energy_efficiency_eia861",
+            "green_pricing_eia861": "green_pricing_eia861",
+            "mergers_eia861": "mergers_eia861",
+            "net_metering_customer_fuel_class_eia861": "net_metering_customer_fuel_class_eia861",
+            "net_metering_misc_eia861": "net_metering_misc_eia861",
+            "non_net_metering_customer_fuel_class_eia861": "non_net_metering_customer_fuel_class_eia861",
+            "non_net_metering_misc_eia861": "non_net_metering_misc_eia861",
+            "operational_data_revenue_eia861": "operational_data_revenue_eia861",
+            "operational_data_misc_eia861": "operational_data_misc_eia861",
+            "reliability_eia861": "reliability_eia861",
+            "utility_data_nerc_eia861": "utility_data_nerc_eia861",
+            "utility_data_rto_eia861": "utility_data_rto_eia861",
+            "utility_data_misc_eia861": "utility_data_misc_eia861",
+            "utility_assn_eia861": "utility_assn_eia861",
+            "balancing_authority_eia861": "balancing_authority_eia861",
+            "balancing_authority_assn_eia861": "balancing_authority_assn_eia861",
+            # ferc714
+            "respondent_id_ferc714": "respondent_id_ferc714",
+            "demand_hourly_pa_ferc714": "demand_hourly_pa_ferc714",
+        }
+
+        for table_name, method_name in table_method_map.items():
+            if hasattr(PudlTabl, method_name):
+                logger.warning(
+                    f"Automatically generated PudlTabl method {method_name} overrides explicitly defined class method. One of these should be deleted."
+                )
+
+            # Create method called asset_name that will read the asset from DB
+            self.__dict__[method_name] = partial(
+                self._get_table_from_db,
+                table_name=table_name,
+                resource=Resource.from_id(table_name),
+            )
+
+    def _get_table_from_db(self, table_name: str, resource: Resource) -> pd.DataFrame:
+        """Grab output table from PUDL DB.
+
+        Args:
+            table_name: Name of table to get.
+            resource: Resource metadata used to enforce schema on table.
+        """
+        return pd.concat(
+            [
+                resource.enforce_schema(df)
+                for df in pd.read_sql(table_name, self.pudl_engine, chunksize=100_000)
```

Comment:
We'll need to ensure that this query respects values of `self.start_date` and `self.end_date` or time and memory intensive unexpected behavior will result.  Ella already has a solution for this in the branch I took over from her, so I can add it in, unless this breaks the CI.

Reply:
That'd be great if you could add that in!